server/services/survey.service.js

```diff
@@ -0,0 +1,60 @@
+const Path = require('path-parser');
+const { URL } = require('url');
+
+module.exports = class SurveyService {
+
+  async updateSurveys(data) {
+    try {
+
+      const uniqueServeys =
+          data.map(this.buildSurveyDataFromBody)
+              .filter(this.filterOutNullAndUndefined)
+              .filter(this.filterByUniqueEmailAndSurveyId);
+
+      for (let survey of uniqueServeys) {
+        await Survey.updateOne(
+            {
+              _id: survey.surveyId,
+              recipients: {
+                $elemMatch: {email: survey.email, responded: false}
+              }
+            },
+            {
+              $inc: {[survey.choice]: 1},
+              $set: {'recipients.$.responded': true},
+              lastResponded: new Date()
+            }
+        ).exec();
+      }
+
+      return { success: true }
+    } catch(error) {
+      return error;
```

Comment:
And here if there was an error, I return that error. You wouldn't want to do this in a real production app, as it could expose details about your code or database that you wouldn't want the public to know, but for this example it's fine, and will actually let you know theres an error, unlike the original code which would fail silently.